scheduler/src/clients/amqp-client.ts

```diff
@@ -0,0 +1,71 @@
+import amqp from 'amqplib'
+
+const AMQP_EXCHANGE = process.env.AMQP_EXCHANGE || 'ods_global'
+const AMPQ_NOTIFICATION_EXECUTION_TOPIC = process.env.AMQP_NOTIFICATION_EXECUTION_TOPIC || 'notification.execution.*'
+const AMQP_URL = process.env.AMQP_URL!
+
+
+let channel: amqp.Channel
+let initialized = false
+
+export async function init(): Promise<void> {
+  if (!initialized) {
+    try {
+      const connection = await connect()
+      channel = await initChannel(connection)
+      initialized = true
+    } catch (error) {
+      console.error(`Error initializing the AMQP Client: ${error}`)
+    }
+  } else {
+    console.log('AmqpClient initialization aborted: Already initialized.')
+    return
+  }
+}
+
+async function connect(): Promise<amqp.Connection> {
+  try {
+    const connection = await amqp.connect(AMQP_URL)
```

Comment:
Having a retry machanism would be nice here! (see storage-mq or notification)

Reply:
good point